app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/DataSetInterface.kt

```diff
@@ -53,4 +54,46 @@ interface DatasetInterface {
      */
     suspend fun getCategories(): Set<Category>
 
+    /**
+     * Retrieves the representative picture of the specified category
+     *
+     * @param category - the category whose representative picture we want to retrieve
+     * @return the representative picture associated to the specified category
+     * @throws IllegalArgumentException if the dataset does not contain the specified category
+     */
+    suspend fun getRepresentativePicture(category: Category): FirestoreCategorizedPicture
+
+    /**
+     * Retrieves all the pictures categorized with the specified category
+     *
+     * @param category - the category whose pictures we want to retrieve
+     * @return the pictures categorized with the specified category
+     * @throws IllegalArgumentException if the dataset does not contain the specified category
+     */
+    suspend fun getAllPictures(category: Category): Set<CategorizedPicture>
+
+    /**
+     * Adds a representative picture to the category. If there is already a representative picture assigned it will be overwritten.
+     *
+     * @param picture - the picture to put as a representative
+     * @param category - the category whose representative picture we want to change
+     * @throws IllegalArgumentException if the dataset does not contain the specified category
+     */
+    suspend fun putRepresentativePicture(picture: android.net.Uri, category: Category)
```

Comment:
This duplicates the functionality of putPicture. Couldn't we putPicture, then assign the resulting picture as representative on the category? KISS

Reply:
Yes, an overload with `putRepresentativePicture(picture: CategorizedPicture)`. This way the representative can be chosen from an already existing picture or be taken from the camera.